utils/serial_tools.py

```diff
@@ -0,0 +1,37 @@
+import json
+import time
+
+import supervisor
+import usb_cdc
+from adafruit_hid.keyboard import Keyboard
+from adafruit_hid.keyboard_layout_us import KeyboardLayoutUS
+from adafruit_hid.keycode import Keycode
+from usb_hid import devices
+
+keyboard = Keyboard(devices)
+keyboard_layout = KeyboardLayoutUS(keyboard)
+
+
+def get_serial_data(is_json: bool = True):
+    if usb_cdc.data.in_waiting > 0:
+        data = usb_cdc.data.readline()
+        if len(data) > 0:
+            usb_cdc.data.reset_input_buffer()
+            return json.loads(data)
+    return {}
+
+
+def trigger_serial_script():
+    keyboard.press(Keycode.F13)
```

Comment:
This should be a configurable setting. 

There may be value in using a `.env` file to be used to configure such settings on the MacroPad.